src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = date.split(fromFormat[3]);
+  const newFormat = fromFormat.slice(0, 3);
+  const newToFormat = toFormat.slice(0, 3);
+  const format = {};
+  const result = [];
+
+  newFormat.forEach((key, index) => {
+    format[key] = newDate[index];
+  });
+
+  newToFormat.forEach((key) => {
```

Comment:
you don't mutate arrays, so there is no need to make a copy, not critical, just FYI